integration-tests/src/test/scala/org/constellation/TransactionsProcessingSpec.scala

```diff
@@ -0,0 +1,144 @@
+
+  "sending transactions from single src address" in {
+    // given
+    val kp = getKeyPair
+    val (src, dst) = (Wallet.getAddress(kp), config.txGen.dst)
+
+    val srcBalances = getBalances(src)
+    srcBalances should containTheSameElements
+
+    val dstBalances = getBalances(dst)
+    dstBalances should containTheSameElements
+    logger.info(s"Initial balances: src address = $src, src balance = ${srcBalances.head}; dst address = $dst, dst balance = ${dstBalances.head}")
+
+    val lastTxRefs = getLasTxRefs(src)
+    lastTxRefs should containTheSameElements
+
+    // when
+    val txs = generateAndSendTxs(kp, lastTxRefs.head)
+    logger.info(s"Sent transactions $txs")
+
+    // then
+    val targetSrcBalance = srcBalances.head - (config.txGen.amount + config.txGen.fee) * config.txGen.countTotal
+    val targetDstBalance = dstBalances.head + config.txGen.amount * config.txGen.countTotal
+
+    eventually {
+      getBalances(src) should contain only targetSrcBalance
+      getBalances(dst) should contain only targetDstBalance
+    }
+  }
+
+  private def getKeyPair =
+    KeyStoreUtils
+      .keyPairFromStorePath[IO](
+        config.txGen.keyPair.keystore,
+        "alias",
+        config.txGen.keyPair.storepass.toCharArray,
+        config.txGen.keyPair.keypass.toCharArray
+      )
+      .value
+      .flatMap {
+        case Right(value) => value.pure[IO]
+        case Left(err)    => err.raiseError[IO, KeyPair]
+      }
+      .unsafeRunSync()
+
+  private def generateAndSendTxs(kp: KeyPair, ltr: LastTransactionRef): List[String] = {
+    val firstTx = mock[Transaction]
+    firstTx.hash shouldReturn ltr.prevHash
+    firstTx.ordinal shouldReturn ltr.ordinal
+
+    val src = Wallet.getAddress(kp)
+
+    def generateTxs(prevTx: Transaction): Stream[IO, Transaction] = {
+      val tx = TransactionExt.createTransaction(prevTx.some, src, config.txGen.dst, config.txGen.amount, kp, (config.txGen.fee.toDouble * 1e-8).some)
+      Stream(tx) ++ generateTxs(tx)
+    }
+
+    evalRequestForHosts[String] {
+      _.flatMap(Stream(_).repeatN(config.txGen.countPerHost))
+        .zipWith(generateTxs(firstTx)) { (host, tx) =>
+          POST(tx.asJson, Uri.unsafeFromString(s"http://$host:9000/transaction"))
+        }
+        .take(config.txGen.countTotal)
+    }
+  }
+
+  private def getBalances(address: String) =
+    evalRequestForHosts[AddressCacheData] {
+      _.map(host => GET(Uri.unsafeFromString(s"http://$host:9000/address/$address")))
+    }.map(_.balance)
+
+  private def getLasTxRefs(address: String) = evalRequestForHosts[LastTransactionRef] {
+    _.map(host => GET(Uri.unsafeFromString(s"http://$host:9000/transaction/last-ref/$address")))
+  }
+
+  private def evalRequestForHosts[T](pipe: Pipe[IO, String, IO[Request[IO]]])(implicit ev: EntityDecoder[IO, T]) =
+    Stream
+      .emits(terraform.instanceIps.value)
+      .through(pipe)
+      .evalMap { req =>
+        BlazeClientBuilder[IO](global).resource.use { httpClient =>
```

Comment:
Should it create separate client for every request?

Reply:
I didn't like that as well, but I was struggling with making clients reusable. I can try again and try to fix it.

Reply:
You can create a Kleisli resource I think

Reply:
I changed it to building a client for a single method call.